src/MPI.jl

```diff
@@ -113,6 +113,20 @@ function __init__()
     end
 
     @static if Sys.isunix()
+        # preload any dependencies of libmpi (if needed, eg. GTL on cray) before
+        # dlopen'ing the MPI library: https://github.com/JuliaParallel/MPI.jl/pull/716
+        preload_enabled = false
+        if isnothing(preloads_env_switch)
+            preload_enabled = true
+        elseif get(ENV, preloads_env_switch, "0") == "1"
+            preload_enabled = true
+        end
+        if preload_enabled
+            for preload in preloads
+                Libdl.dlopen(preload, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
+            end
+        end
```

Comment:
this is where MPItrampoline is erroring: it ends up in a state `preload_enabled == true`, but `preloads == nothing`

Reply:
How about:
```suggestion
        if !isnothing(preloads)
            if isnothing(preloads_env_switch) || get(ENV, preloads_env_switch, "0") == "1"
                for preload in preloads
                    Libdl.dlopen(preload, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
                end
            end
        end
```

Reply:
Actually, this should be redundant altogether, since it should have already been triggered by `MPIPreferences.System.__init__()`